### ISSUE TITLE
Wrong display language for Matter devices

### ISSUE THREAD

**Issue by lboue** [home-assistant/frontend#21371]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Wrong display language for Matter devices.

### Describe the behavior you expected

I'm expecting a display in English like the rest of the interface. Indeed, I've configured my HA profile in English. However, the display is in French.

### Steps to reproduce the issue

1. Configure your HA system in French (Settings->System -> General)
2. Configure your browser in French language
3. Configure your HA profile in English language
4. Browse a Matter device

### What version of Home Assistant Core has the issue?

2024.7.2

### What was the last working version of Home Assistant Core?

N/A

### In which browser are you experiencing the issue with?

Chrome Version 126.0.6478.127 (64 bits)

### Additional information

![image](https://github.com/home-assistant/frontend/assets/938089/7931f794-64f2-4df7-9bd8-86e7717acbf5)

![image](https://github.com/home-assistant/frontend/assets/938089/f17038b9-e55b-4684-a5ff-2f55595b32e9)

**Comment by karwosts** [comment]
I believe entity names are translated based on the system language, those would be set in settings->general, not in an individual user profile language setting.

**Comment by lboue** [comment]
> I believe entity names are translated based on the system language, those would be set in settings->general, not in an individual user profile language setting.

So why are the titles in English and the entities in French?

**Comment by NoRi2909** [comment]
This bug is still fully reproducible in today's October release:

The entity names should follow the user language, not the system language as they are part of the UI. I can reproduce the issue in the Matter integration and in the Sun integration.

Here my system language is set to German, and when I switch the user language to English the labels are stuck in German:

![Screenshot 2024-10-02 21 16 54](https://github.com/user-attachments/assets/b550d044-094d-4dec-a8fb-c54f714aff26)


I can work around this bug by also changing the system language to English and restarting HA.

But this is not the expected behavior and turns a localizer's life into hell by eating up several minutes just to check a single translation in the UI. Those labels need to switch on the fly just like the rest of the UI.

**Comment by joostlek** [comment]
Sun isn't translated afaik. Can you check some more integrations?

**Comment by NoRi2909** [comment]
Sun is fully translated into German. Here in full German compared to above's screenshot with just the entity names stuck in the German system language:

![image](https://github.com/user-attachments/assets/b1afe50a-ebe9-4b29-ab17-342b76a48104)

**Comment by joostlek** [comment]
So this is a not matter specific frontend issue?

**Comment by NoRi2909** [comment]
@joostlek No, not just Matter is affected. Sun and the Xiaomi Mio integration, too:

![image](https://github.com/user-attachments/assets/77a5645e-9d5b-4c04-a589-aeb5fdf6cacf)

Only after changing the system language to English plus restarting those entity names also switch.

**Comment by NoRi2909** [comment]
And also confirmed to affect the WLED integration:

![image](https://github.com/user-attachments/assets/15730680-6035-4202-9685-048dedcb688c)

And that screenshot also confirms what we already saw in the Matter integration:
The items in the popup menus do change with the user language while the labels don't.
Note "Default" and "Off" in the screenshot.

**Comment by joostlek** [comment]
I would maybe recommend creating a new issue which isn't matter specific and explain clearly to what languages HA is set and what the UI looks like. This way there is a good explanation without the added noise of being matter specific

**Comment by NoRi2909** [comment]
@joostlek As suggested, here is the new generalized issue: https://github.com/home-assistant/frontend/issues/22210
